src/CommunityToolkit.Maui/PopupService.cs

```diff
@@ -0,0 +1,127 @@
+﻿using CommunityToolkit.Maui.Views;
+using System.ComponentModel;
+
+namespace CommunityToolkit.Maui;
+
+/// <inheritdoc cref="IPopupService"/>
+public class PopupService : IPopupService
+{
+	readonly IServiceProvider serviceProvider;
+
+	static readonly IDictionary<Type, Type> viewModelToViewMappings = new Dictionary<Type, Type>();
+
+	/// <summary>
+	/// Creates a new instance of <see cref="PopupService"/>.
+	/// </summary>
+	/// <param name="serviceProvider">The <see cref="IServiceProvider"/> implementation.</param>
+	public PopupService(IServiceProvider serviceProvider)
+	{
+		this.serviceProvider = serviceProvider;
+	}
+
+	internal static void AddTransientPopup<TPopupView, TPopupViewModel>(IServiceCollection services)
+		where TPopupView : Popup
+		where TPopupViewModel : INotifyPropertyChanged
+	{
+		viewModelToViewMappings.Add(typeof(TPopupViewModel), typeof(TPopupView));
+
+		services.AddTransient(typeof(TPopupView));
+		services.AddTransient(typeof(TPopupViewModel));
+	}
+
+	/// <inheritdoc cref="IPopupService.ShowPopup{TViewModel}()"/>
+	public void ShowPopup<TViewModel>() where TViewModel : INotifyPropertyChanged =>
+		ShowPopup(GetViewModel<TViewModel>());
+
+	/// <inheritdoc cref="IPopupService.ShowPopup{TViewModel}(TViewModel)"/>
+	public void ShowPopup<TViewModel>(TViewModel viewModel) where TViewModel : INotifyPropertyChanged
+	{
+		ArgumentNullException.ThrowIfNull(viewModel);
+
+		var popup = GetPopup(viewModel);
+
+		GetMainPage().ShowPopup(popup);
+	}
+
+	/// <inheritdoc cref="IPopupService.ShowPopup{TViewModel}(IDictionary{string, object})"/>
+	public void ShowPopup<TViewModel>(IDictionary<string, object> query) where TViewModel : IQueryAttributable
```

Comment:
Let's include `INotifyPropertyChanged` here to match the `TViewModel` constraints on similar methods
```suggestion
	public void ShowPopup<TViewModel>(IDictionary<string, object> query) where TViewModel : IQueryAttributable, INotifyPropertyChanged
```